pkg/jobs/job_scheduler.go

```diff
@@ -351,6 +358,16 @@ func (s *jobScheduler) executeSchedules(
 	return err
 }
 
+// An internal, safety valve setting to revert scheduler execution to distributed mode.
+// This setting should be removed once scheduled job system no longer locks tables for excessive
+// periods of time.
+var schedulerRunsOnSingleNode = settings.RegisterBoolSetting(
+	settings.SystemOnly,
```

Comment:
SystemOnly means tenant code never reads it, so I don't think that's right here. TenantReadOnly is what you want I think

Reply:
We could make the tenant not read it. It's not meaningful to the tenant

Reply:
that was my suggestion in the slack thread: `ForSystemTenant() && setting.Get()`